compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1166,20 +1166,16 @@ public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
 
     @Override
     public void visit(BLangXMLNS xmlnsNode) {
-        String nsURI;
+        String nsURI = "";
```

Comment:
Do we need this change, we can initialise this as null.

Reply:
This was by default initialized to `""` due to this [bug](https://github.com/ballerina-platform/ballerina-lang/issues/42075) and because we decided to handle this separately.